File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution(object):
    def searchRange(self, nums, target):
        left=self.searchindex(nums,target,True)
        right=self.searchindex(nums,target,False)
        return[left,right]

        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
    def searchindex(self,nums,target,dir):
        left=0
        right=len(nums)-1
        result=-1
        while left<=right:
            mid=left+(right-left)//2
            if nums[mid]==target:
                result=mid
                if dir==True:
                    right=mid-1
                else:
                    left=mid+1
            elif nums[mid]<target:
                left=mid+1
            else:
                right=mid-1
        return result

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int left=findindex(nums,target,true);
        int right=findindex(nums,target,false);
           return {left,right};
    }


        int findindex(vector<int>&nums,int target, bool dir){
            int left=0;
            int right=nums.size()-1;
            int result=-1;
            while(left<=right){
                int mid=left+(right-left)/2;
                if(nums[mid]==target){
                    result=mid;
                    if(dir){
                        right=mid-1;
                    }
                    else{
                        left=mid+1;
                    }
                }
                else if(nums[mid]<target){
                    left=mid+1;
                }
                else{
                    right=mid-1;
                }
                }
                return result;
            }
        };
        
    